h2o-py/h2o/grid/infogram_grid.py

```diff
@@ -0,0 +1,44 @@
+import h2o
+
+
+def _train_and_get_models(model_class, x, y, train, **kwargs):
+    from h2o.automl import H2OAutoML
+    from h2o.grid import H2OGridSearch
+
+    model = model_class(**kwargs)
+    model.train(x, y, train)
+    if model_class is H2OAutoML:
+        return [h2o.get_model(m[0]) for m in model.leaderboard["model_id"].as_data_frame(False, False)]
+    elif model_class is H2OGridSearch:
+        return [h2o.get_model(m) for m in model.model_ids]
+    else:
+        return [model]
+
+
+def infogram_grid(infogram, model_class, y, train, **kwargs):
+    """
+    Runs grid over different feature subsets selected by infogram.
+
+    :param infogram: H2OInfogram object
+    :param model_class: H2O Estimator class, H2OAutoML, or H2OGridSearch
+    :param y: response column
+    :param train: training frame
+    :param kwargs: Arguments passed to the constructor of the model_class
+    :return: list of H2O models
+    """
+    from h2o import H2OFrame
```

Comment:
There are two thresholds defaulted to 0.1 I think.  Do you want to be able to grid over those two thresholds as well?  I think one is related to the variable importance value, the other one is the mutual information one.

Reply:
What I currently do is just one infogram which gives me variables with safety index and relevance index. The `infogram_grid` creates subsets of features in the following way: {{safest}, {safest, 2nd safest}, {safest, 2nd safest, 3rd safest},...}. This creates models that are often at least as good as those trained with the same resources on all the features and it also makes the models more interpretable since they often use less features.

I am also considering another way to pick features that would take care of correlated features which doesn't necessarily happen in the current implementation of infogram. 
1. create infogram
2. add the best feature to S
3. calculate conditional mutual information conditioned on the features that is currently being conditioned on plus the features in the set S
4. go to (2)
This way features that are strongly correlated would become less useful once a correlated feature is added to S. And this could lead to a better set of features than just using the infogram since infogram looks only at one feature at a time. Does it make sense to you @wendycwong?

Reply:
So the thresholds are not really used in the infogram_grid, the infogram is used just for ranking the features and subsequent feature subset creation.

Reply:
Yes, it makes sense.  Thank you.  In this case, I would like to recommend the maxr way to choose your predictor subset.  However, you will not be R2 as the determining factor but some other metrics.  However, the way to choose the predictor subsets remains the same.  We can talk more about this offline.